modules/core/src/smithy4s/internals/SchematicDocumentDecoder.scala

```diff
@@ -229,25 +229,18 @@ object SchematicDocumentDecoder extends Schematic[DocumentDecoderMake] {
         ()
     })
 
-  def list[S](fs: DocumentDecoderMake[S]): DocumentDecoderMake[List[S]] =
+  def collection[C[_], S](
+      tag: CollectionTag[C, S],
+      fs: DocumentDecoderMake[S]
+  ): DocumentDecoderMake[C[S]] =
     fs.transform { fa =>
-      DocumentDecoder.instance("List", "Array", false) {
+      DocumentDecoder.instance("Collection", "Array", false) {
```

Comment:
Call upon the [name](https://github.com/disneystreaming/smithy4s/pull/290/files#r913496776) method